apps/src/gamelab/animationListModule.js

```diff
@@ -0,0 +1,605 @@
+    key
+  };
+}
+
+/**
+ * Clone the requested animation, putting the new one directly after the original
+ * in the project animation list.
+ * @param {!AnimationKey} key
+ * @returns {Function}
+ */
+export function cloneAnimation(key) {
+  return (dispatch, getState) => {
+    const state = getState().animationList;
+
+    var onCloneError = function (errorMessage) {
+      dispatch(reportError(`Error copying object ${key}: ${errorMessage}`));
+    };
+
+    // Track down the source animation and its index in the collection
+    var sourceIndex = state.list.indexOf(key);
+    if (sourceIndex < 0) {
+      onCloneError('Animation not found');
+      return;
+    }
+    var sourceAnimation = state.data[key];
+    var newAnimationKey = utils.createUuid();
+
+    /**
+     * Once the cloned asset is ready, call this to add the appropriate metadata.
+     * @param {string} [versionId]
+     */
+    var addClonedAnimation = function (versionId) {
+      dispatch({
+        type: ADD_ANIMATION_AT,
+        index: sourceIndex + 1,
+        key: newAnimationKey,
+        data: Object.assign({}, sourceAnimation, {
+          name: sourceAnimation.name + '_copy', // TODO: better generated names
+          version: versionId
+        })
+      });
+      dispatch(selectAnimation(newAnimationKey));
+      dashboard.project.projectChanged();
+    };
+
+    // If cloning a library animation, no need to perform a copy request
+    if (/^\/blockly\//.test(sourceAnimation.sourceUrl)) {
+      addClonedAnimation();
+    } else {
+      animationsApi.ajax(
+          'PUT',
+          newAnimationKey + '.png?src=' + key + '.png',
+          function success(xhr) {
+            try {
+              var response = JSON.parse(xhr.responseText);
+              addClonedAnimation(response.versionId);
+            } catch (e) {
+              onCloneError(e.message);
+            }
+          },
+          function error(xhr) {
+            onCloneError(xhr.status + ' ' + xhr.statusText);
+          });
+    }
+  };
+}
+
+/**
+ * Set the display name of the specified animation.
+ * @param {string} key
+ * @param {string} name
+ * @returns {{type: ActionType, key: string, name: string}}
+ */
+export function setAnimationName(key, name) {
+  return dispatch => {
+    dispatch({
+      type: SET_ANIMATION_NAME,
+      key,
+      name
+    });
+    dashboard.project.projectChanged();
+  };
+}
+
+/**
+ * Modifies the animation data, capturing changes to its spritesheet.
+ * @param {!AnimationKey} key
+ * @param {object} data - needs a more detailed shape
+ */
+export function editAnimation(key, data) {
+  return dispatch => {
+    dispatch({
+      type: EDIT_ANIMATION,
+      key,
+      data
+    });
+    dashboard.project.projectChanged();
+  };
+}
+
+/**
+ * Delete the specified animation from the project.
+ * @param {!AnimationKey} key
+ * @returns {function}
+ */
+export function deleteAnimation(key) {
+  return dispatch => {
+    animationsApi.ajax(
```

Comment:
TODO: Likewise, we should be deleting immediately and trying to put a delete marker on S3 asynchronously behind the scenes.  In fact, as this is right now I wonder if we could accidentally save an animation _while_ deleting it?

Since we'll eventually need some sort of cleanup job, I wonder if it's worth doing this delete at all?